Failed because of a test failure of a single build:
```
[ERROR] org.apache.jackrabbit.oak.upgrade.cli.SegmentTarToSegmentAzureTest  Time elapsed: 0.29 s  <<< ERROR!

java.lang.RuntimeException: Unable to start docker container: DockerConfig{name=oak-test-azurite}
	at com.arakelian.docker.junit.DockerRule$StatementWithDockerRule.evaluate(DockerRule.java:76)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: java.lang.IllegalStateException: Unable to create container using ContainerConfig{hostname=null, domainname=null, user=null, attachStdin=null, attachStdout=null, attachStderr=null, portSpecs=null, exposedPorts=[10000], tty=null, openStdin=null, stdinOnce=null, env=[executable=blob], cmd=null, image=trekawek/azurite, volumes={}, workingDir=null, entrypoint=null, networkDisabled=false, onBuild=null, labels=null, macAddress=null, hostConfig=HostConfig{binds=null, blkioWeight=null, blkioWeightDevice=null, blkioDeviceReadBps=null, blkioDeviceWriteBps=null, blkioDeviceReadIOps=null, blkioDeviceWriteIOps=null, containerIdFile=null, lxcConf=null, privileged=null, portBindings={10000=[PortBinding{hostIp=0.0.0.0, hostPort=}]}, links=null, publishAllPorts=null, dns=null, dnsOptions=null, dnsSearch=null, extraHosts=null, volumesFrom=null, capAdd=null, capDrop=null, networkMode=null, securityOpt=null, devices=null, memory=null, memorySwap=null, memorySwappiness=null, memoryReservation=null, nanoCpus=null, cpuPeriod=null, cpuShares=null, cpusetCpus=null, cpusetMems=null, cpuQuota=null, cgroupParent=null, restartPolicy=null, logConfig=null, ipcMode=null, ulimits=null, pidMode=null, shmSize=null, oomKillDisable=null, oomScoreAdj=null, autoRemove=null, pidsLimit=null, tmpfs=null, readonlyRootfs=null, storageOpt=null}, stopSignal=null, healthcheck=null, networkingConfig=null}
	at com.arakelian.docker.junit.Container.createContainer(Container.java:449)
	at com.arakelian.docker.junit.Container.start(Container.java:251)
	at com.arakelian.docker.junit.DockerRule$StatementWithDockerRule.evaluate(DockerRule.java:72)
	... 11 more
Caused by: com.spotify.docker.client.exceptions.ImageNotFoundException: Image not found: trekawek/azurite
```

--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/Cursors.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/Cursors.java
@@ -16,7 +16,6 @@
  */
 package org.apache.jackrabbit.oak.plugins.index;
 
-import java.util.Arrays;
 import java.util.Deque;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -342,7 +341,7 @@ public class Cursors {
                         FilterIterators.checkReadLimit(readCount, settings);
                         String caller = IndexUtils.getCaller(this.settings.getIgnoredClassNamesInCallTrace());
                         LOG.warn("Traversed {} nodes with filter {} called by {}; consider creating an index or changing the query" , 
-                                new Object[] {readCount, filter, caller});
+                                readCount, filter, caller);
                     }
 
                     NodeState node = entry.getNodeState();

--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUtils.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUtils.java
@@ -30,7 +30,6 @@ import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.REINDEX_PRO
 import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.TYPE_PROPERTY_NAME;
 import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.UNIQUE_PROPERTY_NAME;
 
-import java.util.Arrays;
 import java.util.Collection;
 import java.util.Map;
 import java.util.Set;
@@ -42,7 +41,6 @@ import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Iterables;
 import com.google.common.collect.Sets;
 import org.apache.jackrabbit.oak.api.PropertyState;
-import org.apache.jackrabbit.oak.api.QueryEngine;
 import org.apache.jackrabbit.oak.api.Tree;
 import org.apache.jackrabbit.oak.api.Type;
 import org.apache.jackrabbit.oak.plugins.index.property.PropertyIndexEditorProvider;
@@ -253,7 +251,7 @@ public final class IndexUtils {
      * actual caller doesn't matter (or cannot be determined clearly). In this case a short message
      * indicating this is returned.
      *
-     * If the ingoredJavaPackages parameter is null or empty, the caller is not looked up, but
+     * If the ignoredJavaPackages parameter is null or empty, the caller is not looked up, but
      * instead it is assumed, that the feature is not configured; in that case a short messages
      * is returned indicating that the feature is not configured.
      *
@@ -270,7 +268,7 @@ public final class IndexUtils {
         final StackTraceElement[] callStack = Thread.currentThread().getStackTrace();
         for (StackTraceElement stackFrame : callStack) {
             final String classAndMethod = stackFrame.getClassName() + "." + stackFrame.getMethodName();
-            if (Stream.of(ignoredJavaPackages).noneMatch(pkg -> classAndMethod.startsWith(pkg))) {
+            if (Stream.of(ignoredJavaPackages).noneMatch(classAndMethod::startsWith)) {
                 return classAndMethod;
             }
         }

--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/property/PropertyIndexPlan.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/property/PropertyIndexPlan.java
@@ -24,7 +24,6 @@ import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.DECLARING_N
 import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.INDEX_CONTENT_NODE_NAME;
 import static org.apache.jackrabbit.oak.plugins.index.IndexConstants.PROPERTY_NAMES;
 
-import java.util.Arrays;
 import java.util.List;
 import java.util.Set;
 

--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettingsService.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettingsService.java
@@ -82,7 +82,7 @@ public class QueryEngineSettingsService {
                 name="Fully qualified class names to ignore when finding caller",
                 description="If non-empty the query engine logs the query statement plus the java package "
                         + "which executed this query. This java package is the first package in the call trace"
-                        + "which does not not start with the any of the provided fully qualfied class names (packagename + classname)"
+                        + "which does not  start with any of the provided fully qualified class names (packagename + classname)"
                 )
         String[] ignoredClassNamesinCallTrace() default {};
 

--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettingsService.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineSettingsService.java
@@ -81,10 +81,10 @@ public class QueryEngineSettingsService {
         @AttributeDefinition(
                 name="Fully qualified class names to ignore when finding caller",
                 description="If non-empty the query engine logs the query statement plus the java package "
-                        + "which executed this query. This java package is the first package in the call trace"
+                        + "which executed this query. This java package is the first package in the call trace "
                         + "which does not  start with any of the provided fully qualified class names (packagename + classname)"
                 )
-        String[] ignoredClassNamesinCallTrace() default {};
+        String[] ignoredClassNamesInCallTrace() default {};
 
     }
 
@@ -130,7 +130,7 @@ public class QueryEngineSettingsService {
             logMsg(QUERY_FAIL_TRAVERSAL, QueryEngineSettings.OAK_QUERY_FAIL_TRAVERSAL);
         }
 
-        queryEngineSettings.setIgnoredClassNamesInCallTrace(config.ignoredClassNamesinCallTrace());
+        queryEngineSettings.setIgnoredClassNamesInCallTrace(config.ignoredClassNamesInCallTrace());
 
         boolean fastQuerySizeSysProp = QueryEngineSettings.DEFAULT_FAST_QUERY_SIZE;
         boolean fastQuerySizeFromConfig = config.fastQuerySize();

--- a/oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/IndexUtilsTest.java
+++ b/oak-core/src/test/java/org/apache/jackrabbit/oak/plugins/index/IndexUtilsTest.java
@@ -28,6 +28,12 @@ import static org.apache.jackrabbit.oak.plugins.memory.EmptyNodeState.EMPTY_NODE
 import static org.junit.Assert.*;
 
 public class IndexUtilsTest {
+    
+    // all relevant package TLDs
+    private static final String[] ALL_CLASSES_IGNORED = new String[] {"org", "com", "sun", "jdk", "java"};
+    
+    // all packages used with Oak
+    private static final String[] OAK_CLASSES_IGNORED = new String[] {"org.apache.jackrabbit", "java.lang", "sun.reflect", "jdk"};
 
     @Test
     public void asyncName() throws Exception {
@@ -41,4 +47,14 @@ public class IndexUtilsTest {
         assertEquals("async3", IndexUtils.getAsyncLaneName(builder.getNodeState(), "/fooIndex"));
     }
 
+    @Test
+    public void getCaller() {
+        assertNotNull(IndexUtils.getCaller(null));
+        assertNotNull(IndexUtils.getCaller(new String[0]));
+        
+        assertEquals("(internal)",IndexUtils.getCaller(ALL_CLASSES_IGNORED));
+        
+        String caller = IndexUtils.getCaller(OAK_CLASSES_IGNORED);
+        assertTrue(caller.startsWith("org.junit.runners"));
+    }  
 }

